server/datastore/mysql/activities_test.go

```diff
@@ -10,10 +10,25 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestActivityUsernameChange(t *testing.T) {
+func TestActivity(t *testing.T) {
```

Comment:
@chiiph  as discussed in #1821 , this is how I see "test suites" being implemented in standard Go tests/sub-tests:

* the "suite" is the top-level test function (for mysql I created one by test file, with the name being the name of the file/entity).
* the suite setup/teardown is any code executed and deferred (respectively) before the sub-tests (can also be `t.Cleanup` instead of defer, as is the case in `CreateMySQLDS`).
* I used a table-driven approach to run the suite's tests, as it tends to minimize code repetition.
* Per-test setup/teardown is just the code inside the `t.Run` closure that runs before calling the test function.
* If some test has specific setup/teardown, of course it can simply be done in that test's function.

In this case I did not need any shared state so I didn't use a struct to hold the suite's tests, but it is a possibility (I could've used a struct and store e.g. the datastore there so its methods would've accessed it there instead of as an argument).

Reply:
Yeah, makes sense and it's simple enough.